### ISSUE TITLE
Null pointer - java.security.AccessController.doPrivileged

### ISSUE THREAD

**Issue by h1z1** [chatty/chatty#102]
Never saw this one before or it's been hidden ? 

```
[2016-10-27 11:36:36 INFO] Got 0 (limit: 100) followed streams. [chatty.util.api.StreamInfoManager/requestResultFollows]                                              
[2016-10-27 11:39:20 INFO] Log: total: 4703 / queue size (avg: 0, max: 0) / errors: 0 [chatty.util.chatlog.LogWriter/outputStats]                                     
[2016-10-27 11:39:55 INFO] GET (v3): https://api.twitch.tv/kraken/streams/followed?limit=100&offset=0 (using authorization) [chatty.util.api.TwitchApiRequest/getUrl] 
[2016-10-27 11:39:56 INFO] GOT (200, 196, gzip): https://api.twitch.tv/kraken/streams/followed?limit=100&offset=0 (using authorization) [chatty.util.api.TwitchApi/req
uestResult]                                                                                                                                                           
[2016-10-27 11:39:56 INFO] Got 0 (limit: 100) followed streams. [chatty.util.api.StreamInfoManager/requestResultFollows]                                              
[2016-10-27 11:43:08 INFO] [FFZ] Request: http://cdn.frankerfacez.com/script/event.json [chatty.util.UrlRequest/run]                                                  



Error Report // Chatty Version 0.8.3 / JDK8 / Ja
va: 1.8.0_91 (Oracle Corporation) OS: Linux (4.5.0-1.el7.elrepo.x86_64/amd64) / [Memory] total: 1,344,000 used: 1,000,489 free: 343,510 max: 7,296,512 [Uptime] 12h 0m 1s

11:43:08 [FFZ] Request: https://api.frankerfacez.com/v1/set/22546
11:43:08 [FFZ] Response (200, 5436): https://api.frankerfacez.com/v1/set/22546
11:43:08 Changed 'BloodTrailWeen'/FFZ emote size: 180x32 -> 100x17
11:43:08 [FFZ] (22546, global): 17 emotes received.
11:43:08 Unhandled Exception:
java.lang.NullPointerException
	at chatty.util.api.Emoticons.removeEmoticons(Emoticons.java:156)
	at chatty.util.api.Emoticons.updateEmoticons(Emoticons.java:126)
	at chatty.gui.MainGui$76.run(MainGui.java:3197)
	at java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$500(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.run(Unknown Source)


[2016-10-27 11:43:08 INFO] [FFZ] Response (200, 112): http://cdn.frankerfacez.com/script/event.json [chatty.util.UrlRequest/run]                                      
[2016-10-27 11:43:08 INFO] [FFZ] Request: https://api.frankerfacez.com/v1/set/22546 [chatty.util.UrlRequest/run]                                                      
[2016-10-27 11:43:08 INFO] [FFZ] Response (200, 5436): https://api.frankerfacez.com/v1/set/22546 [chatty.util.UrlRequest/run]                                         
[2016-10-27 11:43:08 WARNING] Changed 'BloodTrailWeen'/FFZ emote size: 180x32 -> 100x17 [chatty.util.api.Emoticon/<init>]                                             
[2016-10-27 11:43:08 INFO] [FFZ] (22546, global): 17 emotes received. [chatty.util.ffz.FrankerFaceZ/parseResult]                                                      
[2016-10-27 11:43:08 SEVERE] java.lang.NullPointerException                                                                                                           
        at chatty.util.api.Emoticons.removeEmoticons(Emoticons.java:156)                                                                                              
        at chatty.util.api.Emoticons.updateEmoticons(Emoticons.java:126)                                                                                              
        at chatty.gui.MainGui$76.run(MainGui.java:3197)                                                                                                               
        at java.awt.event.InvocationEvent.dispatch(Unknown Source)                                                                                                    
        at java.awt.EventQueue.dispatchEventImpl(Unknown Source)                                                                                                      
        at java.awt.EventQueue.access$500(Unknown Source)                                                                                                             
        at java.awt.EventQueue$3.run(Unknown Source)                                                                                                                  
        at java.awt.EventQueue$3.run(Unknown Source)                                                                                                                  
        at java.security.AccessController.doPrivileged(Native Method)                                                                                                 
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)                                                              
        at java.awt.EventQueue.dispatchEvent(Unknown Source)                                                                                                          
        at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)                                                                                        
        at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)                                                                                           
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)                                                                                        
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source)                                                                                                    
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source)                                                                                                    
        at java.awt.EventDispatchThread.run(Unknown Source)                                                                                                           
 [chatty.ErrorHandler/uncaughtException]                                                                                                                              
[2016-10-27 11:43:14 INFO] Request: http://chatty.github.io/news.json [chatty.util.UrlRequest/run]                                                                    
[2016-10-27 11:43:14 INFO] Response (200, 380): http://chatty.github.io/news.json [chatty.util.UrlRequest/run]                                                        
[2016-10-27 11:43:15 INFO] GET (v3): https://api.twitch.tv/kraken/streams/followed?limit=100&offset=0 (using authorization) [chatty.util.api.TwitchApiRequest/getUrl] 
[2016-10-27 11:43:16 INFO] GOT (200, 196, gzip): https://api.twitch.tv/kraken/streams/followed?limit=100&offset=0 (using authorization) [chatty.util.api.TwitchApi/req
uestResult]                                                                                                                                               
```

Grrr... Interesting it's pulling chatty.github.io/news.json  .. that should be disabled!

**Comment by tduva** [comment]
It happening because of the FFZ Halloween Emotes. There's an error in trying to remove a set of global emotes in order to update them. Since it's usually just channel-specific emotes that need to be removed I guess I didn't test it properly.

What do you mean it should be disabled?

**Comment by h1z1** [comment]
Hm.. Thanks.  Kinda figured that give the URL but wasn't expecting the exception :)   FFZ and BTTV emotes have been disabled for now.  Forgot when I updated to turn them off again.


Unless I missed the option somewhere, update checking is disabled afik.  The only place I can think of where announcements are displayed is in Help->Announcements, click refresh - no?

**Comment by tduva** [comment]
Unfortunately I didn't expect the exception either. D:

The Announcements aren't the same as checking for updates. They are intended for important stuff that everyone should know about, so there is no GUI setting for it. You can use "/set newsAutoRequest 0" to disable it.

**Comment by h1z1** [comment]
Hmm.  OK this is annoying now, crashed again overnight.  Third party Emotes are _disabled_  (they don't show up in chat either) yet:



Error Report // Chatty Version 0.8.3 / JDK8 / Java: 1.8.0_91 (Oracle Corporation) OS: Linux (4.5.0-1.el7.elrepo.x86_64/amd64) / [Memory] total: 850,432 used: 213,409 free: 637,022 max: 7,296,512 [Uptime] 12h 0m 0s

00:32:25 [FFZ] Request: https://api.frankerfacez.com/v1/set/22546
00:32:25 [FFZ] Response (200, 5440): https://api.frankerfacez.com/v1/set/22546
00:32:25 Changed 'BloodTrailWeen'/FFZ emote size: 180x32 -> 100x17
00:32:25 [FFZ] (22546, global): 17 emotes received.
00:32:25 Unhandled Exception:
java.lang.NullPointerException
	at chatty.util.api.Emoticons.removeEmoticons(Emoticons.java:156)
	at chatty.util.api.Emoticons.updateEmoticons(Emoticons.java:126)
	at chatty.gui.MainGui$76.run(MainGui.java:3197)
	at java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$500(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.run(Unknown Source)


06:32:25 [FFZ] Request: https://api.frankerfacez.com/v1/set/22546
06:32:25 [FFZ] Response (200, 5440): https://api.frankerfacez.com/v1/set/22546
06:32:25 Changed 'BloodTrailWeen'/FFZ emote size: 180x32 -> 100x17
06:32:25 [FFZ] (22546, global): 17 emotes received.
06:32:25 Unhandled Exception:
java.lang.NullPointerException
	at chatty.util.api.Emoticons.removeEmoticons(Emoticons.java:156)
	at chatty.util.api.Emoticons.updateEmoticons(Emoticons.java:126)
	at chatty.gui.MainGui$76.run(MainGui.java:3197)
	at java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$500(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.WaitDispatchSupport$2.run(Unknown Source)
	at java.awt.WaitDispatchSupport$4.run(Unknown Source)
	at java.awt.WaitDispatchSupport$4.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.awt.WaitDispatchSupport.enter(Unknown Source)
	at java.awt.Dialog.show(Unknown Source)
	at java.awt.Component.show(Unknown Source)
	at java.awt.Component.setVisible(Unknown Source)
	at java.awt.Window.setVisible(Unknown Source)
	at java.awt.Dialog.setVisible(Unknown Source)
	at chatty.gui.components.ErrorMessage.show(ErrorMessage.java:222)
	at chatty.gui.MainGui$93.run(MainGui.java:3939)
	at java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$500(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.run(Unknown Source)


??

**Comment by gmt2001** [comment]
Noticing the same error. It is caused by this line https://github.com/chatty/chatty/blob/master/src/chatty/util/api/Emoticons.java#L173

It is attempting to use `update.roomToRemove` when it is null as an array index in a `get` method. Apparently you accounted for the possibility of `update.roomToRemove` being null near the top of the method but not on this line. Just needs an `if (update.roomToRemove != null)` wrapper around that line.

**Comment by h1z1** [comment]
Care to propose a patch ?  As simple as it sounds, my java knowledge is fairly limited.  I'm not entirely sure why the remove fails.  What I can say is this has happened ~4 times today despite having FFZ and BTTV DISABLED.   There should be NO reason for it to call home (this part bothers me more then the error itself).

**Comment by tduva** [comment]
Don't bother with patches, this is already being taken care of.

**Comment by broadstone** [comment]
This appears to be fixed by d1573b137a4efe9017a1f042bfcf2a1f689e864d

**Comment by h1z1** [comment]
Closing.  Thanks tduva.  Updated this morning, no crashes since.

**Comment by h1z1** [comment]
Happened again on both  patched  and 0.8.3 clients...


Error Report // Chatty Version 0.8.4 / JDK8 / Java: 1.8.0_91 (Oracle Corporation) OS: Linux (4.5.0-1.el7.elrepo.x86_64/amd64) / [Memory] total: 949,760 used: 556,334 free: 393,425 max: 7,296,512 [Uptime] 12h 0m 0s

00:37:45 [FFZ] Request: https://api.frankerfacez.com/v1/set/22546
00:37:45 [FFZ] Response (200, 5440, gzip): https://api.frankerfacez.com/v1/set/22546
00:37:45 Changed 'BloodTrailWeen'/FFZ emote size: 180x32 -> 100x17
00:37:45 [FFZ] (22546, global): 17 emotes received.
00:37:45 Unhandled Exception:
java.lang.NullPointerException
	at chatty.util.api.Emoticons.removeEmoticons(Emoticons.java:173)
	at chatty.util.api.Emoticons.updateEmoticons(Emoticons.java:143)
	at chatty.gui.MainGui$82.run(MainGui.java:3352)
	at java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$500(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.run(Unknown Source)

**Comment by h1z1** [comment]
Maybe unrelated but java is consuming almost a gig of _resident_ memory (not shared).  That is on the 0.8.4 instance, just sitting in one channel with no one else.


VmPeak:	13080456 kB
VmSize:	13079704 kB
VmLck:	       0 kB
VmPin:	       0 kB
VmHWM:	 1032028 kB
VmRSS:	 1031216 kB
RssAnon:	 1007980 kB
RssFile:	   22748 kB
RssShmem:	     488 kB
VmData:	 1607752 kB
VmStk:	     136 kB
VmExe:	       4 kB
VmLib:	   22488 kB
VmPTE:	    2560 kB
VmPMD:	      44 kB
VmSwap:	       0 kB

?
